autosklearn/automl.py

```diff
@@ -228,32 +227,6 @@ def fit(
             only_return_configuration_space=only_return_configuration_space,
         )
 
-    # TODO this is very old code which can be dropped!
-    def fit_automl_dataset(self, dataset, metric, load_models=True):
-        self._stopwatch = StopWatch()
-        self._backend.save_start_time(self._seed)
-
-        name = os.path.basename(dataset)
-        self._stopwatch.start_task(name)
-        self._start_task(self._stopwatch, name)
-        self._dataset_name = name
-
-        self._logger = self._get_logger(name)
-        self._logger.debug('======== Reading and converting data ==========')
-        # Encoding the labels will be done after the metafeature calculation!
-        self._data_memory_limit = float(self._ml_memory_limit) / 3
-        loaded_data_manager = CompetitionDataManager(
-            dataset, max_memory_in_mb=self._data_memory_limit)
-        loaded_data_manager_str = str(loaded_data_manager).split('\n')
-        for part in loaded_data_manager_str:
-            self._logger.debug(part)
-
-        return self._fit(
-            datamanager=loaded_data_manager,
-            metric=metric,
-            load_models=load_models,
-        )
-
     def fit_on_datamanager(self, datamanager, metric, load_models=True):
```

Comment:
Hey, I think this function can go, too.

Reply:
Also, if these two additional functions are removed, do you think it still makes sense to have a `fit` and an `_fit` function? Without looking at the code too much, I don't think it does. If you agree, could you please make them a single `fit` function?